encorelab/SCORE-Client
# Manage groups: display, add, delete, change membership

## Changes
- Added manage group page in teacher management tool
- In the new manage group page, added the following features:
  - add new group with/without initial members
  - display existing groups
  - edit existing group name
  - edit existing group membership
  - delete existing groups

## Test
- First checkout and pull from the "manage-groups" branch in SCORE-API, to make sure that you have the updated backend code for deleting and retrieving groups
```
SCORE-API $ git checkout manage-groups
SCORE-API $ git pull
```
- Verify that you can do all of the above grouping actions as a teacher 
- Try branch based on groups "History", "Science", and "Math"
  - Download project and import into your SCORE instance: [3.zip](https://github.com/encorelab/SCORE-Client/files/6872990/3.zip)
  - Set up run, and add students to it. Launch the run as student, and go to step 2.1, which is the branching point.
  - Open another tab and log in as teacher.
  - As teacher, create three groups: "History", "Science", and "Math". Add students to the group
  - Verify that the students are branched according to the group they are assigned to.
